logdna/logdna.py

```diff
@@ -1,72 +1,81 @@
 import sys
 import time
 import json
-from .configs import defaults
 import logging
-import requests
-import threading
 import socket
-from threading import Timer
+import threading
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+from .configs import defaults
+from .utils import sanitize_meta, get_ip
+
 logger = logging.getLogger(__name__)
 
 class LogDNAHandler(logging.Handler):
     def __init__(self, key, options={}):
-        self.buf = []
-        self.secondary = []
         logging.Handler.__init__(self)
+
         self.key = key
-        self.hostname = options['hostname'] if 'hostname' in options else socket.gethostname()
-        self.ip = options['ip'] if 'ip' in options else self.get_ip()
-        self.mac = options['mac'] if 'mac' in options else None
-        self.level = options['level'] if 'level' in options else 'info'
-        self.verbose = str(options['verbose']).lower() if 'verbose' in options else 'true'
-        self.app = options['app'] if 'app' in options else ''
-        self.env = options['env'] if 'env' in options else ''
-        self.url = options['url'] if 'url' in options else defaults['LOGDNA_URL']
-        self.setLevel(logging.DEBUG)
-        self.exceptionFlag = False
-
-        self.tags = []
-        if 'tags' in options and isinstance(options['tags'], list):
-            self.tags.extend(options['tags'])
-        self.max_length = True
-        if 'max_length' in options:
-            self.max_length = options['max_length']
-        self.index_meta = False
-        if 'index_meta' in options:
-            self.index_meta = options['index_meta']
-        self.include_standard_meta = False
-        if 'include_standard_meta' in options:
-            self.include_standard_meta = options['include_standard_meta']
-        self.flushLimit = defaults['FLUSH_BYTE_LIMIT']
-        self.bufByteLength = 0
+        self.buf = []
+        self.secondary = []
+        self.exception_flag = False
+        self.buf_byte_length = 0
         self.flusher = None
+        self.max_length = defaults['MAX_LINE_LENGTH']
+        self.connection_retries = 5
+        self.retry_backoff_factor = 0.5
+
+        self.hostname = options.get('hostname', socket.gethostname())
+        self.ip = options.get('ip', get_ip())
+        self.mac = options.get('mac', None)
```

Comment:
`None` is the default for `get()` so you don't need that. But it's fine to keep.